app/assets/javascripts/index.js

```diff
@@ -238,11 +238,37 @@ $(document).ready(function () {
     L.marker(center, {icon: OSM.getUserIcon()}).addTo(map);
   });
 
+  var isMobile = {
+    Windows: function() {
+        return /IEMobile/i.test(navigator.userAgent);
+    },
+    Android: function() {
+        return /Android/i.test(navigator.userAgent);
+    },
+    BlackBerry: function() {
+        return /BlackBerry/i.test(navigator.userAgent);
+    },
+    iOS: function() {
+        return /iPhone|iPad|iPod/i.test(navigator.userAgent);
+    },
+    any: function() {
+        return (isMobile.Android() || isMobile.BlackBerry() || isMobile.iOS() || isMobile.Windows());
+    }
+  };
+
+  function url() {
```

Comment:
Fine, but call it `remoteEditUrl` or something and pass it the query string so the whole of the url logic can be in one place...